js/styles.js

```diff
@@ -0,0 +1,197 @@
+      // case the style has slightly higher priority than the link
+      if (tagName === "LINK" && newTagName === "STYLE") {
+        // The style is allowed to be inserted after the link, so continue
+        return false;
+      }
+
+      if (tagName === newTagName) {
+        // Two elements of the same tag with the same priority are not
+        // allowed - raise a warning about it, but add them anyway
+        console.error(
+          "Interwiki: encountered two " +
+            (tagName === "LINK" ? "themes" : "CSS styles") +
+            " with the same priority (" +
+            priority +
+            ") - result may not be as expected"
+        );
+        // Fall back to insertion
+      }
+
+      // Otherwise (if old is STYLE and new is LINK, or if they are equal),
+      // fall back to insertion
+    }
+
+    // Otherwise, if the new priority is less than the current priority (or
+    // after falling back from the equality case), insert the new element
+    // before the one with higher priority
+    stylingElement.insertBefore(newStylingElement);
+    return true;
+  });
+
+  if (!wasInserted) {
+    // The element would not have been inserted if it is the first, or if
+    // its priority is greater than all existing priorities
+    document.head.appendChild(newStylingElement);
+  }
+}
+
+/**
+ * Constructs and returns a function that checks if a given HTML element
+ * (assumed to be either a link or a style element) is a match for an
+ * element that would have been created for the given priority and value.
+ *
+ * @param {Number} priority
+ * @param {String} value
+ */
+function duplicatesStyle(priority, value) {
+  /**
+   * @param {HTMLLinkElement | HTMLStyleElement} styleElement
+   * @returns {Boolean}
+   */
+  var isDuplicate = function (styleElement) {
+    if (Number(styleElement.getAttribute("data-priority")) !== priority) {
+      return false;
+    }
+    if (styleElement.tagName === "LINK") {
+      return styleElement.getAttribute("href") === value;
+    }
+    if (styleElement.tagName === "STYLE") {
+      return styleElement.innerText === value;
+    }
+    return false;
+  };
+  return isDuplicate;
+}
+
+/**
+ * Constructs a URL pointing to the expected location of a CSS stylesheet.
+ *
+ * The theme pointer may be one of the following:
+ * - A full URL to a CSS stylesheet.
+ * - The fullname of a page on the current wiki, whose /code/1 endpoint
+ *   will be parsed as a CSS stylesheet.
+ * - The fullname of a page on the current wiki specifying a code block
+ *   number (of the form <fullname>/code/n), where the nth code block on
+ *   that page will be parsed as a CSS stylesheet.
+ *
+ * @param {String} theme - An indicator of where a CSS stylesheet can be
+ * found.
+ */
+function urlFromTheme(theme) {
+  // If the theme is already a full URL, return it
+  if (theme.indexOf("http") === 0 || theme.indexOf("//") === 0) return theme;
```

Comment:
Having single-line `if`s go from inline to next line is confusing imo

Reply:
Fair enough - my rationale is that <code>return theme</code> is practically a no-op so an inline return sort of makes sense, but the next return concats some strings so it's not a no-op so an inline return makes less sense, even though it would fit on the line. But I agree in retrospect that it does look a bit weird like this.